dashboard/src/reducers/namespace.ts

```diff
@@ -26,11 +26,24 @@ const namespaceReducer = (
   action: NamespaceAction | LocationChangeAction | AuthAction,
 ): INamespaceState => {
   switch (action.type) {
+    case getType(actions.namespace.receiveNamespace):
+      if (!state.namespaces.includes(action.payload.metadata.name)) {
+        return {
+          ...state,
+          namespaces: state.namespaces.concat(action.payload.metadata.name),
+          error: undefined,
+        };
+      }
+      return state;
     case getType(actions.namespace.receiveNamespaces):
       return { ...state, namespaces: action.payload };
     case getType(actions.namespace.setNamespace):
       return { ...state, current: action.payload, error: undefined };
     case getType(actions.namespace.errorNamespaces):
+      // Ignore error listing namespaces since those are expected
```

Comment:
Does it depend on the error? We expect a 4xx if the user can't get namespaces, but we don't expect other errors?

Reply:
not really, at least not at this moment, we are just ignoring any error. I know that if we receive a 5xx we should probably do something but that would mean that the k8s API is not responding which the user would notice in some other way? In any case, this wouldn't be a regression.